LoopKitUI/Views/BasalScheduleEntryTableViewCell.swift

```diff
@@ -0,0 +1,336 @@
+        }
+    }
+
+    var startTime: TimeInterval = 0 {
+        didSet {
+            updateStartTimeSelection()
+            updateDateLabel()
+        }
+    }
+
+    var selectedStartTime: TimeInterval {
+        let row = picker.selectedRow(inComponent: Component.time.rawValue)
+        return startTimeForTimeComponent(row: row)
+    }
+
+    var value: Double = 0 {
+        didSet {
+            updateValuePickerWith(value)
+            updateValueLabel()
+        }
+    }
+
+    var unitString: String? {
+        get {
+            return unitLabel.text
+        }
+        set {
+            unitLabel.text = newValue
+        }
+    }
+
+    var isPickerHidden: Bool {
+        get {
+            return picker.isHidden
+        }
+        set {
+            picker.isHidden = newValue
+            pickerHeightConstraint.constant = newValue ? 0 : pickerExpandedHeight
+        }
+    }
+
+    private lazy var startOfDay: Date = {
+        return Calendar.current.startOfDay(for: Date())
+    }()
+
+    var isReadOnly = false
+
+    override func awakeFromNib() {
+        super.awakeFromNib()
+
+        pickerExpandedHeight = pickerHeightConstraint.constant
+
+        setSelected(true, animated: false)
+        updateDateLabel()
+    }
+
+    override func setSelected(_ selected: Bool, animated: Bool) {
+        super.setSelected(selected, animated: animated)
+
+        if selected && !isReadOnly {
+            isPickerHidden = !isPickerHidden
+        }
+    }
+
+    private lazy var dateFormatter: DateFormatter = {
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateStyle = .none
+        dateFormatter.timeStyle = .short
+
+        return dateFormatter
+    }()
+
+    lazy var valueNumberFormatter: NumberFormatter = {
+        let formatter = NumberFormatter()
+        formatter.numberStyle = .decimal
+        formatter.minimumFractionDigits = 1
+
+        return formatter
+    }()
+
+    lazy var wholeNumberFormatter: NumberFormatter = {
+        let formatter = NumberFormatter()
+        formatter.numberStyle = .decimal
+        formatter.minimumFractionDigits = 0
+        return formatter
+    }()
+
+    lazy var fractionalNumberFormatter: NumberFormatter = {
+        let formatter = NumberFormatter()
+        formatter.numberStyle = .decimal
+        formatter.minimumFractionDigits = 3
+        formatter.maximumIntegerDigits = 0
+        return formatter
+    }()
+
+    private func startTimeForTimeComponent(row: Int) -> TimeInterval {
+        return minimumStartTime + pickerInterval * TimeInterval(row)
+    }
+
+    private func stringForStartTime(_ time: TimeInterval) -> String {
+        let date = startOfDay.addingTimeInterval(time)
+        return dateFormatter.string(from: date)
+    }
+
+    func updateDateLabel() {
+        dateLabel.text = stringForStartTime(startTime)
+    }
+
+    func validate() {
+        if delegate?.validateBasalScheduleEntryTableViewCell(self) == true
+        {
```

Comment:
`{` on preceding line? Plus, no need for `== true`.

Reply:
[nit] ping

Reply:
The result of this expression is an optional.  If there is no delegate for validation, we don't consider the value invalid.